src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc

```diff
@@ -106,96 +106,131 @@ void CreateInteractionChoiceSetRequest::Run() {
     return;
   }
 
-  Result::eType result = CheckChoiceSet(app);
+  Result::eType result = CheckChoiceSet();
   if (Result::SUCCESS != result) {
     SendResponse(false, result);
```

Comment:
@LevchenkoS you Check choice set function return bool. You cast it to enum. Confusing. 
Also you send response in CheckChoiceSet function and send response after it's call that depends of i't result ( double response). 

Please avoid sending response from CheckChoiceSet